src/allmydata/test/no_network.py

```diff
@@ -401,10 +402,7 @@ def _record_webports_and_baseurls(self):
                                 for c in self.g.clients]
 
     def get_clientdir(self, i=0):
-        return self.g.clients[i].basedir
-
-    def set_clientdir(self, basedir, i=0):
-        self.g.clients[i].basedir = basedir
+        return self.g.clients[i].config._basedir
```

Comment:
It would be nice to avoid this `config._basedir`.  `NoNetworkGrid` could probably figure out the basedir for a particular client number by re-doing the calculation at the beginning of `NoNetworkGrid.make_client` since that seems deterministic...  So that logic could go into a `NoNetworkGrid.get_client_basedir` (or something) method, called from both `make_client` and `get_clientdir`?

Reply:
I think I can just nuke `get_clientdir` in favour of `get_client_config` since it's only used in one place anyway